iroh/tests/cmd/get_ipfs_path_success.out/QmP8jTG1m9GSDJLCbeWhVSVgEzCPPwXRdCRuJtQ5Tz9Kc9/b

```diff
@@ -0,0 +1 @@
+hello
```

Comment:
https://github.com/n0-computer/iroh/blob/d3fb9b4677170ea70595e607078a6ce1687d760c/iroh/src/fixture.rs#L31-L45

`fixture_get` sends a stream of 1) the wrapped directory, 2) a directory called "a" 3) a file called "b"
(If "b" is supposed to be nested below "a", then the path to "b" should be `RelativePathBuf::from_path("a/b")`)

Which means that there should be 2 entries below the `get_ipfs_path_success.out/QmP8jTG1m9GSDJLCbeWhVSVgEzCPPwXRdCRuJtQ5Tz9Kc9` folder, an empty folder "a", and the file "b" that reads "hello".

When I run `TRYCMD=dump cargo test --test cli_tests`
The dump output does show the directory `get_ipfs_path_success.out/QmP8jTG1m9GSDJLCbeWhVSVgEzCPPwXRdCRuJtQ5Tz9Kc9/a`, which leads me to believe that it should be creating it. But it isn't flagging as an error.  Unless this ignores empty directories for some reason?

Reply:
This just has me worried that we aren't actually testing the behavior that happens on the actual command line that we want to test.

Reply:
trycmd is definitely finicky sometimes, will investigate empty folder behavior. May adjust the test to deal with nested streams - a separate unit test for saving the stream can then confirm creation.

Reply:
Aha, I was slightly confused about this as the empty directories do exist for me in the `out` directories, but I think this has to do with `git` just not adding empty directories and me missing that. So for me everything looks right but in the PR the directories are missing. This indeed means `trycmd` seems to tolerate missing (empty?) directories. I will adjust the tests to avoid empty directories first and we'll look into `trycmd` later.

Reply:
Okay, `trycmd` seems to not only tolerate an `.out` with a directory entirely missing, but even a *file* entirely missing. The docs say: 

"When present, each file in this directory will be compared to generated or modified files."

so it only compares things that are both there, and doesn't warn that it's missing. Basically it only compares what's present in `.out` and nothing else. That's very surprising and I created an issue for it: https://github.com/assert-rs/trycmd/issues/149

Reply:
Update: what I wrote here isn't exactly right. If a file is present in `.out` and not in the output, `trycmd` does trip up. It's the other way around that's tricky: it doesn't detect *extra* stuff is generated and is missing from `.out`.

Reply:
I've now made sure I don't generate empty directories and ensured there's a file in all of them to compare. I'd still like a trycmd strict mode that fails, but it'll do for now.